finality-aleph/src/sync/service.rs

```diff
@@ -104,6 +102,13 @@ where
         let broadcast_ticker = Ticker::new(BROADCAST_PERIOD, BROADCAST_COOLDOWN);
         let (justifications_for_sync, justifications_from_user) = mpsc::unbounded();
         let (block_requests_for_sync, block_requests_from_user) = mpsc::unbounded();
+        let metrics = match Metrics::new(metrics_registry) {
+            Ok(metrics) => metrics,
+            Err(e) => {
+                warn!(target: LOG_TARGET, "Failed to create metrics: {}.", e);
+                Metrics::noop()
```

Comment:
Maybe just fail? That creates a possibility of running a node without metrics for some time without noticing it.

Reply:
Hah, running without metrics should be the _easiest_ thing to notice though, especially if you have alerting for them. :D

I feel like returning an error here requires writing a lot of boilerplate, while the actual noop fallback is quite acceptable for metrics.

Reply:
I don't understand. This constructor already returns some error and when it occurs we just immediately panic. You can extend it or switch with `anyhow::Result`? Feel free to resolve this comment.

Reply:
The error it returns is from `Handler`, which cannot fail on metrics, so I would have to create a separate error just for this. We don't currently use `anyhow`. Also, this would be inconsistent with the changes in `clique`, and there we don't return any error, so probably another one would have to be added there.